### PR TITLE
chore: migrate last files from pages folder to app folder

### PR DESCRIPTION
**Pull request by rmonnier9**
- Refacto
- Détails :
  - Migration des derniers fichiers de pages à app :
  -> 4 API routes
  -> 5 pages

### CODE REVIEW

File: app/(header-default)/donnees/api-entreprises/page.tsx [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
-      noIndex={false}
-      canonical="https://annuaire-entreprises.data.gouv.fr/donnees/acceder-par-api"
-    />
+export const metadata: Metadata = {
+  title: 'API Recherche d’entreprises & API Entreprise',
+  alternates: {
+    canonical:
+      'https://annuaire-entreprises.data.gouv.fr/donnees/acceder-par-api',
+  },
+  robots: 'noindex, nofollow',
```
Comment (XavierJp) [inline app/(header-default)/donnees/api-entreprises/page.tsx:10]:
```suggestion
  robots: 'index, follow',
```

File: app/(header-default)/donnees/sources/page.tsx [hunk 2]
```diff
@@ -1,32 +1,28 @@
...
-interface IProps {
-  allAdministrations: IAdministrationMetaData[];
-  allAPI: { [apiSlug: string]: IAPIMonitorMetaData };
-}
+export const metadata: Metadata = {
+  title: 'Sources des données utilisées dans l’Annuaire des Entreprises',
+  alternates: {
+    canonical: 'https://annuaire-entreprises.data.gouv.fr/donnees/sources',
+  },
+  robots: 'noindex, nofollow',
```
Comment (XavierJp) [inline app/(header-default)/donnees/sources/page.tsx:16]:
```suggestion
  robots: 'index, follow',
```

File: app/(header-default)/donnees/sources/page.tsx [hunk 3]
```diff
@@ -1,32 +1,28 @@
...
-  <>
-    <Meta
-      title="Sources des données utilisées dans l’Annuaire des Entreprises"
-      canonical="https://annuaire-entreprises.data.gouv.fr/donnees/sources"
-    />
+export default function DataSourcesPage() {
+  // TO CHECK
+  const allAdministrations = Object.values(administrationsMetaData).filter(
+    (administration) => administration.dataSources.length > 0
+  );
```
Comment (XavierJp) [inline app/(header-default)/donnees/sources/page.tsx:23]: What needs to be checked ?

File: app/(header-default)/partager/page.tsx [hunk 4]
```diff
@@ -1,15 +1,18 @@
...
-import { NextPageWithLayout } from 'pages/_app';
+import { Metadata } from 'next';
 
-const Partager: NextPageWithLayout = () => {
+export const metadata: Metadata = {
+  title: 'Réutiliser ou partager l’Annuaire des Entreprises',
+  alternates: {
+    canonical: 'https://annuaire-entreprises.data.gouv.fr/partager',
+  },
+  robots: 'noindex, nofollow',
```
Comment (XavierJp) [inline app/(header-default)/partager/page.tsx:10]:
```suggestion
  robots: 'index, follow',
```

**Comment by rmonnier9** [comment]
Rather than writing "robots: 'index, follow", we can remove the line ?

File: app/(header-default)/donnees/sources/page.tsx (hunk 3 above)
Comment (rmonnier9) [reply app/(header-default)/donnees/sources/page.tsx:23]: If it was working !

**Comment by rmonnier9** [comment]
> > Rather than writing "robots: 'index, follow", we can remove the line ?
> 
> I guess we can, but I advised against. From my past experience, it is best to have explicit indexing declarations as Jonathan frequently spot some new regressions

But most of our pages, don't have any "robot" property, shoud we add it everywhere ?
Maybe we can do that in an other PR ?

**Pull request merged**